test/partition.test.js

```diff
@@ -185,32 +186,60 @@ describe('cloudant - partitioned db', () => {
           },
         },
       });
-      db.automigrate('Product', () => {
-        Product.create(SEED_DATA, function(err) {
+      db.automigrate('Product', (err)=> {
+        if (err) return done(err);
+        Product.create(SEED_DATA, done);
+      });
+    });
+
+    it('find all records by partition key from option', (done) => {
+      Product.find({where: {tag: 'food'}}, {partitionKey: 'toronto'},
+        (err, results) => {
           if (err) return done(err);
-          Product.find({where: {tag: 'food'}}, {partitionKey: 'toronto'},
-            (err, results) => {
-              if (err) return done(err);
-              should.exist(results);
-              results.length.should.equal(2);
-              const resultTaggedFood = _.filter(results,
-                function(r) {
-                  return r.tag === 'food';
-                });
-              resultTaggedFood.length.should.equal(2);
-              done();
+          should.exist(results);
+          results.length.should.equal(2);
+          const resultTaggedFood = _.filter(results,
+            function(r) {
+              return r.tag === 'food' && r.city === 'toronto';
             });
+          resultTaggedFood.length.should.equal(2);
+          done();
+        });
+    });
+
+    it('find all records by partition key from property', (done) => {
+      Product.find({where: {tag: 'food', city: 'toronto'}},
+        (err, results) => {
+          if (err) return done(err);
+          should.exist(results);
+          results.length.should.equal(2);
+          const resultTaggedFood = _.filter(results,
+            function(r) {
+              return r.tag === 'food' && r.city === 'toronto';
+            });
+          resultTaggedFood.length.should.equal(2);
+          done();
+        });
+    });
+
+    it('partition key in options overrides property for find all', (done) => {
```

Comment:
Do you mean the `partitionKey` in option changes `city: {type: String, isPartitionKey: true}` to `city: {type: String, isPartitionKey: false}`?